Makefile

```diff
@@ -1,6 +1,8 @@
 SHELL = /bin/bash
+TAG = $(shell git describe)$(shell [[ $$(git status --porcelain) = "" ]] || echo -dirty)
```

Comment:
TAG is parsed by `git describe`, which returns nothing if there is no annotated tag.